inbox/util/threading.py

```diff
@@ -16,25 +19,24 @@ def fetch_corresponding_thread(db_session, namespace_id, message):
     # to a message always has a similar subject. This is only
     # right 95% of the time.
     clean_subject = cleanup_subject(message.subject)
+
+    # XXX: It is much faster to sort client-side by message ID, we therefore
+    # use `contains_eager` and `outerjoin` instead of `joinedload` to fetch
+    # the messages by thread in no particular order. We also use a limit to
+    # avoid scanning too many / large threads.
     threads = db_session.query(Thread). \
         filter(Thread.namespace_id == namespace_id,
                Thread._cleaned_subject == clean_subject). \
+        outerjoin(Message, Thread.messages). \
```

Comment:
Do we really need an *outer* join here? We don't really have a need for threads that contain no messages, do we?

Reply:
Same question. If this helps with the query plan we should just document that so we remember in the future.

Reply:
It matches the previous query generated by SQLAlchemy, and in general how loads work (if you fetch threads with messages you'd expect to have all the threads, even if they have no messages). If you specifically wanted to include threads without messages, you could use `join`. But since all threads have messages anyways, why change the type of join?